.travis.yml

```diff
@@ -105,13 +117,25 @@ matrix:
         sources: *pkg_src_zeromq_ubuntu14
         packages:
         - *pkg_deps_common
-  
+#  - env: BUILD_TYPE=cmake DO_CLANG_FORMAT_CHECK=1 CLANG_FORMAT=clang-format-5.0
+# For non-cmake users, there is an autotools solution with a bit more overhead
+# to have dependencies ready and pass configure script before making this check):
+  - env: BUILD_TYPE=clang-format-check CLANG_FORMAT=clang-format-5.0
+    os: linux
+    dist: trusty
+    addons:
+      apt:
+        sources:
+        - llvm-toolchain-trusty-5.0
+        packages:
+        - clang-5.0
+        - clang-format-5.0
 
 before_install:
 - if [ "$TRAVIS_OS_NAME" == "osx" -a "$BUILD_TYPE" == "android" ] ; then brew install binutils ; fi
 - if [ "$TRAVIS_OS_NAME" == "osx" -a "$BUILD_TYPE" == "valgrind" ] ; then brew install valgrind ; fi
-- sudo apt-get remove liblog4cplus-dev
-- sudo apt-get autoremove
+#- sudo apt-get remove liblog4cplus-dev
```

Comment:
Because we build our own fork (so far with customized packaging). We don't use the upstream one "as is" either.

Reply:
Note that above we (should) no longer install a packaged version either, and it is not part of the default OS footprint AFAIK. And less operations translates to faster iterations (why did we install to remove in the first place)?

Although the comment-away rather than line deletion should serve to more easily see and revert if this change does explode in a build ;) So it is sort of deprecation, can be finally removed some time later when it is known-safe.

Reply:
I'm agreed that we want use our fork, but if you don't have this 2 lines, the ci_build.sh (line 385-386) will found an already present package log4cplus-dev, so he will not get our fork.

Reply:
You did notice that the older `.travis.yml` installed the package first, then removed it, right? And that now it does not waste time doing either?

Looking at the green tests (whatever was used - sufficed) and into Travis logs like https://api.travis-ci.org/v3/job/363098696/log.txt we can see that it actually builds our fork to test against.

Reply:
Ok, i understand now.
Thanks for the explanations.